app/services/device_tracking/list_device_events.rb

```diff
@@ -0,0 +1,9 @@
+module DeviceTracking
+  class ListDeviceEvents
+    def self.call(user, device_id)
+      user_id = user.id
+      return [] unless Device.where(user_id: user_id, device_id: device_id)
+      Event.where(user_id: user_id, device_id: device_id).order(created_at: :desc)
```

Comment:
It might make sense to create a general pattern for pagination of long lists.

An example elsewhere that could give some hints is the Relay GraphQL design: https://facebook.github.io/relay/graphql/connections.htm